specification/_types/aggregations/pipeline.ts

```diff
@@ -18,13 +18,14 @@
  */
 
 import { Sort } from '@global/search/_types/sort'
+import { Dictionary } from '@spec_utils/Dictionary'
 import { Name, Field } from '@_types/common'
 import { integer, double, float } from '@_types/Numeric'
 import { Script } from '@_types/Scripting'
 import { Aggregation } from './Aggregation'
 
 export class PipelineAggregationBase extends Aggregation {
-  buckets_path?: BucketsPath
+  buckets_path?: string | Dictionary<string, string>
```

Comment:
From what I can see in the ES code base, the type of `bucket_path` heavily depends on the enclosing aggregation type: it can be a simple `string`, a simple `Dictionary<string, string>` or also a `string[]`. So it will have to be defined for each individual pipeline aggregation.

This some significant work, so will be revisited in a different PR.

In the meantime, can you also add the `string[]` union member and leave a TODO comment that this need to be revisited and specialized for each pipeline aggregation?